### PR TITLE
Add Image feature extraction on Hub as a task

### PR DESCRIPTION
**Pull request by merveenoyan**
This PR adds image feature extraction to Hub.
Icon looks like below 👇 

Screenshot 2023-12-07 at 15 49 53

### CODE REVIEW

File: packages/widgets/src/lib/components/Icons/IconImageFeatureExtraction.svelte [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" {...$$props}
+	><path
+		fill="currentColor"
+		d="M5 21q-.825 0-1.412-.587T3 19V5q0-.825.588-1.412T5 3h5v2H5v14h14v-5.35l2 2V19q0 .825-.587 1.413T19 21H5Zm1-4l3-4l2.25 3l3-4L18 17H6Zm15.55-3.6l-3.1-3.1q-.525.35-1.125.525T16.05 11q-1.85 0-3.15-1.312T11.6 6.5q0-1.875 1.313-3.187T16.1 2q1.875 0 3.188 1.313T20.6 6.5q0 .675-.2 1.3t-.5 1.15L22.95 12l-1.4 1.4ZM16.1 9q1.05 0 1.775-.725T18.6 6.5q0-1.05-.725-1.775T16.1 4q-1.05 0-1.775.725T13.6 6.5q0 1.05.725 1.775T16.1 9Z"
+	/></svg
+>
```
Comment (mishig25) [inline packages/widgets/src/lib/components/Icons/IconImageFeatureExtraction.svelte:6]:
```suggestion
<script lang="ts">
	export let classNames = "";
</script>

<svg
	class={classNames}
	xmlns="http://www.w3.org/2000/svg"
	xmlns:xlink="http://www.w3.org/1999/xlink"
	aria-hidden="true"
	fill="currentColor"
	focusable="false"
	role="img"
	width="1em"
	height="1em"
	preserveAspectRatio="xMidYMid meet"
	viewBox="0 0 24 24"
>
	<path
		fill="currentColor"
		d="M5 21q-.825 0-1.412-.587T3 19V5q0-.825.588-1.412T5 3h5v2H5v14h14v-5.35l2 2V19q0 .825-.587 1.413T19 21H5Zm1-4l3-4l2.25 3l3-4L18 17H6Zm15.55-3.6l-3.1-3.1q-.525.35-1.125.525T16.05 11q-1.85 0-3.15-1.312T11.6 6.5q0-1.875 1.313-3.187T16.1 2q1.875 0 3.188 1.313T20.6 6.5q0 .675-.2 1.3t-.5 1.15L22.95 12l-1.4 1.4ZM16.1 9q1.05 0 1.775-.725T18.6 6.5q0-1.05-.725-1.775T16.1 4q-1.05 0-1.775.725T13.6 6.5q0 1.05.725 1.775T16.1 9Z"
	/>
</svg>
```

**Review by gary149 — APPROVED** [review]
ok

**Review by osanseviero — CHANGES_REQUESTED** [review]
This would add a filter/tag that leads to 0 models on the Hub (see https://huggingface.co/models?other=image-feature-extraction&sort=trending). We should first:
* add the tag to the corresponding `transformers` and `timm` repos. Which models would have this tag?
* explore adding the `pipeline` to `transformers`

From a quick look, most models in https://huggingface.co/models?pipeline_tag=feature-extraction&other=vision are candidates to be `image-feature-extraction`. These models are usually automatically tagged, so step 3 in https://www.notion.so/huggingface2/How-do-pipelines-get-determined-on-the-Hub-2fdc2f3ae1a44690b43e28adbe326d11 needs to be done. This also needs to align with the `transformers` team first to see if there's a need for this pipeline/task  cc @LysandreJik @amyeroberts 

There is a very related discussion on https://github.com/huggingface/transformers/issues/20830#issuecomment-1370626529 about this, we should check if `feature-extraction` works for these

**Comment by merveenoyan** [comment]
@osanseviero  I made a script to run over timm models so I will be autotagging them. There's a lot of backbones on Hub, they're not explored well among other modalities, this is discussed in the issue linked to this PR.

**Comment by osanseviero** [comment]
>  can do the same for transformers feature-extraction models that are for vision

Please carefully review the resources/Notion linked above. **We should not open PRs for the vision-tagged `transformers` models.** The `pipeline_tag` in `transformers` models is automatically detected, so actually we need to add `ImageFeatureExtractionPipeline` there and update the mapping, which will automatically update the tags for `transformers` repos

**Comment by merveenoyan** [comment]
@osanseviero 

>  so actually we need to add ImageFeatureExtractionPipeline there and update the mapping, which will automatically update the tags for transformers repos

whom should I confirm with to do so?

**Review by osanseviero — APPROVED** [review]
Thanks!

**Pull request merged**